crates/viewer/re_component_ui/src/variant_uis/redap_uri_button.rs

```diff
@@ -32,21 +32,46 @@ pub fn redap_uri_button(
 
     let uri = RedapUri::from_str(url_str)?;
 
-    //TODO(#10036): we should provide feedback if the URI is already loaded, e.g. have "go to" instead of "open"
-    if ui
-        .button("Open")
-        .on_hover_ui(|ui| {
-            ui.label(uri.to_string());
-        })
-        .clicked()
-    {
-        let url = if ui.input(|i| i.modifiers.command) {
-            egui::OpenUrl::new_tab(uri)
+    let loaded_recording_id = ctx.storage_context.bundle.entity_dbs().find_map(|db| {
+        if db
+            .data_source
+            .as_ref()
+            .is_some_and(|source| source.redap_uri().as_ref() == Some(&uri))
+        {
+            Some(db.store_id())
         } else {
-            egui::OpenUrl::same_tab(uri)
-        };
+            None
+        }
+    });
+
+    if let Some(loaded_recording_id) = loaded_recording_id {
+        let response = ui.button("Switch to").on_hover_ui(|ui| {
+            ui.label("This recording is already loaded. Click to switch to it.");
+        });
+
+        if response.clicked() {
+            // Show it:
+            ctx.command_sender()
+                .send_system(SystemCommand::SetSelection(
+                    re_viewer_context::Item::StoreId(loaded_recording_id),
+                ));
+        }
+    } else {
+        let response = ui.button("Open").on_hover_ui(|ui| {
+            ui.label(uri.to_string());
+        });
+
+        if response.middle_clicked() {
+            ui.ctx().open_url(egui::OpenUrl::new_tab(uri));
+        } else if response.clicked() {
+            let url = if ui.input(|i| i.modifiers.any()) {
+                egui::OpenUrl::new_tab(uri)
+            } else {
+                egui::OpenUrl::same_tab(uri)
+            };
 
-        ui.ctx().open_url(url);
+            ui.ctx().open_url(url);
+        }
```

Comment:
there gotta be a utility for this in egui, right..?

Reply:
it will soon: https://github.com/emilk/egui/pull/7093 😆